### PR TITLE
[THRIFT-2820] Implement an alternative Transport for JS using the AngularJS $http-service

### PR DESCRIPTION
**Pull request by uvwxy**
I added the target --gen js:angular to generate angularjs targeted thrift service clients. The thrift.js has been modified simliar to the --gen js:jquery target.

### CODE REVIEW

**Comment by uvwxy** [comment]
See https://issues.apache.org/jira/browse/THRIFT-2820?focusedCommentId=14389241&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-14389241 for explanation

**Comment by henrique** [comment]
Hi Paul,
Thanks for the PR. It looks pretty good, but have tried to remove the jQuery dependency?
As you mentioned on your comments, I also think it make more sense if we didn't need it but I am not completely sure if you can run $http only with jqLite...

**Comment by uvwxy** [comment]
Hi Henrique,

the dependecy is only needed when $http is not initialised correctly. In this case I fall back to the jqRequest mechanism to complete the tests.

File: lib/js/test/test-angular.html [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+<head>
+  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+  <title>Thrift Javascript Bindings: Unit Test</title>
+
+  <script src="build/js/thrift.js"         type="text/javascript" charset="utf-8"></script>
+  <script src="gen-js/ThriftTest_types.js" type="text/javascript" charset="utf-8"></script>
+  <script src="gen-js/ThriftTest.js"       type="text/javascript" charset="utf-8"></script>
+
+  <!-- AngularJS + Dependency-->
+  <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.3/jquery.min.js"></script>
```
Comment (henrique) [inline lib/js/test/test-angular.html]: Could we remove the jq dependency here?

Comment (uvwxy) [inline lib/js/test/test-angular.html]: The dependency is required to run the tests that do not use the angular js service.

**Comment by jeking3** [comment]
This pull request is being closed due to lack of activity.  
You can re-open it after rebasing on the current master if you would like it to be reconsidered.

**Pull request closed without merging**